lib/core/src/persist/mod.rs

```diff
@@ -375,6 +375,21 @@ impl Persister {
             .optional()?)
     }
 
+    pub fn get_payment_by_destination(&self, destination: &str) -> Result<Option<Payment>> {
+        Ok(self
+            .get_connection()?
+            .query_row(
+                &self.select_payment_query(
+                    Some("(rs.invoice = ?1 OR ss.invoice = ?1 OR cs.claim_address = ?1 OR pd.destination = ?1)"),
```

Comment:
It seems that a payment may have multiple destinations? This means that get_payment_by_destination may return same result for different destinations? for example for a claim_address and bip21 that was parsed as input?

Reply:
@roeierez Currently there's no case where this can happen. Specifically, here's the cases:
- Receive Swap/Chain Swap: No payment details set, no conflict can occur
- Send swap using Boltz: No payment details set, no conflict can occur
- Direct Send/Send "swap" with invoice MRH: We persist data as a Liquid payment, therefore there is no data in the swap tables
- Generic non-SDK payment (as of #474): We persist the tx's output's script pubkey in the payment details. We have no data in the swap tables.

So in all cases the swap/details are mutually exclusive, meaning there's no way for two inputs to return the same payment.

Reply:
@dangeross do you think it makes sense instead of a string to use an enum so the query will be more intuitive and search will look up for the specific identifier?
```
PaymentDestination {
  Lightning(bolt11),
  Liquid(address),
  Bitcoin(address)
}
```

Reply:
@roeierez Yeah, it does make it more intuitive and fits in with other functions of the SDK.